src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -3494,17 +3496,12 @@ public String getKeyspaceReplicationInfo(String keyspaceName)
     public Set<InetAddressAndPort> getLiveRingMembers(boolean excludeDeadStates)
     {
         Set<InetAddressAndPort> ret = new HashSet<>();
-        for (InetAddressAndPort ep : Gossiper.instance.getLiveMembers())
+        for (InetAddressAndPort ep : getTokenMetadata().getAllRingMembers())
```

Comment:
why do we need these changes?

Reply:
This change is part of the original port and looks like it was committed as part of the tokenmetadata changes and not as part of the custom failure detection changes. Since it's in the original port I think we should leave it in.